src/Puppeteer.js

```diff
@@ -144,4 +146,23 @@ module.exports = class {
   createBrowserFetcher(options) {
     return new BrowserFetcher(this._projectRoot, options);
   }
+
+  /**
+   * @param {string} name
+   * @param {!Function} queryFunction
+   */
+  registerCustomQueryFunction(name, queryFunction) {
```

Comment:
bikeshed: `Function` sounds a little generic. How about `Handler`?

Reply:
I was aware that this file was likely going over to TS at some point soon, and therefore would get a "proper" signature at that point. AFAICT you can't easily specify JSDoc for function params, but maybe I missed something.

Reply:
👍 on this - let's leave this and be much more explicit and define nicer types when we migrate this file to TS

Reply:
I was talking about the method name (`registerCustomQueryFunction`), not the types

Reply:
Well that's changed now, too! 🥳